Gems/PhysX/Code/Editor/Source/Components/Conversion/CollidersPrefabConversion.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Asset/AssetManager.h>
+#include <AzCore/Console/IConsole.h>
+
+#include <AzFramework/API/ApplicationAPI.h>
+
+#include <EditorColliderComponent.h>
+#include <EditorShapeColliderComponent.h>
+#include <EditorRigidBodyComponent.h>
+#include <EditorStaticRigidBodyComponent.h>
+
+#include <Editor/Source/Components/Conversion/PrefabConversionUtils.h>
+
+namespace PhysX
+{
+    void FixPrefabsWithColliderComponents(const AZ::ConsoleCommandContainer& commandArgs);
+
+    AZ_CONSOLEFREEFUNC(
+        "ed_physxFixPrefabsWithColliderComponents",
+        FixPrefabsWithColliderComponents,
+        AZ::ConsoleFunctorFlags::Null,
+        "Finds entities with collider components and no rigid bodies and fixes them by adding a static rigid body component.");
+
+    bool AddStaticRigidBodyToPrefabEntity(
+        Utils::PrefabInfo& prefabInfo,
+        AzToolsFramework::Prefab::PrefabDomValue& entityPrefab)
+    {
+        AZ::Entity entity;
+        Utils::PrefabEntityIdMapper prefabEntityIdMapper;
+
+        if (!Utils::LoadPrefabEntity(prefabEntityIdMapper, entityPrefab, entity))
+        {
+            AZ_Warning(
+                "PhysXColliderConversion",
+                false,
+                "Unable to load entity '%s' from prefab '%s'.",
+                entity.GetName().c_str(),
+                prefabInfo.m_prefabFullPath.c_str());
+            return false;
+        }
+
+        if (!entity.CreateComponent<EditorStaticRigidBodyComponent>())
+        {
+            AZ_Warning(
+                "PhysXColliderConversion",
+                false,
+                "Failed to create static rigid body component for entity '%s' in prefab '%s'.",
+                entity.GetName().c_str(),
+                prefabInfo.m_prefabFullPath.c_str());
+            return false;
+        }
+
+        if (!Utils::StorePrefabEntity(prefabEntityIdMapper, prefabInfo.m_template->GetPrefabDom(), entityPrefab, entity))
+        {
+            AZ_Warning(
+                "PhysXColliderConversion",
+                false,
+                "Unable to store entity '%s' into prefab '%s'.",
+                entity.GetName().c_str(),
+                prefabInfo.m_prefabFullPath.c_str());
+            return false;
+        }
+
+        return true;
+    }
+
+    void FixPrefabPhysXColliders(Utils::PrefabInfo& prefabInfo)
+    {
+        bool prefabModified = false;
+        for (auto* entity : Utils::GetPrefabEntities(prefabInfo.m_template->GetPrefabDom()))
+        {
+            const auto entityComponents = Utils::GetEntityComponents(*entity);
+
+            const bool rigidBodyMissing = AZStd::none_of(
```

Comment:
Would it not work to just do something like entity->FindComponent<EditorRigidBodyComponent>()?

Reply:
I misunderstood this originally because I was thinking entity was an AZ::Entity*, but could it be worth writing a FindComponent utility function?

Reply:
I've only wrote utils that are used and in this case I would still use the `AZStd:none_of` and `AZStd::any_of` to have into account it's searching for multiple types. But in the future if it' needed we can totally add it.